modules/tutorials/pages/schema-design-guide.adoc

```diff
@@ -1 +1,267 @@
+In a *friendship relationship*, for example, if *Person A* is friends with *Person B*, then *Person B* is also friends with *Person A*.
+This means the relationship works in both directions, and we don’t need to specify a direction for it.
+
+image::Undirected edges.png[]
+
+The following schema creates an undirected edge for friendship:
+
+[source.wrap,gsql]
+----
+CREATE EDGE Is_Friend_Of (FROM Person, TO Person)
+----
+
+*Directed Edge (Work Relationship)*
+
+*Directed edges* are used for *one-way relationships* where the direction of the relationship matters.
+For instance, in a *work relationship*, Person A might work for Company B, but Company B does not work for Person A.
+This clearly defines that the relationship flows only from Person A to Company B.
+
+image::Directed Edge.png[]
+
+The following schema creates a directed edge for *work relationship*:
+
+[source.wrap,gsql]
+----
+CREATE EDGE Works_For (FROM Person, TO Company)
+----
+
+*Reversed Edge (Parent-Child Relationship)*
+
+A *reversed edge* allows you to define both directions of a relationship.
+For example, in a *parent-child relationship*, we might define the primary edge as *"parent_of"* (Parent → Child).
+However, to easily traverse the relationship in the opposite direction (from Child to Parent), we can use a *reversed edge* with a different name, like "child_of" (Child → Parent). This makes it easy to query in either direction.
+
+image::Reversed Edge.png[]
+
+The following schema creates *reversed edges* for a *parent-child* relationship:
+
+[source.wrap,gsql]
+----
+CREATE EDGE Parent_Of (FROM Person, TO Person)
+CREATE EDGE Child_Of (FROM Person, TO Person)
+----
+
+==== Multiple Relationships Between Two Vertices
+
+For situations where different types of relationships occur between two vertices (e.g., purchases and reviews, or purchases and returns), you can represent each relationship as a discriminated edge.
+
+The figure below shows a 'Person' vertex connected to a 'Product' vertex by two different edges: 'purchased' and 'returned'. Each edge clearly indicates a different type of interaction.
+
+image::Multiple relationship between two vertices.png[]
+
+The following scheme creates discriminated edges to represent different types of *events* (e.g., *purchase and return*):
+
+[source.wrap,gsql]
+----
+CREATE EDGE Purchase_Event (FROM Person, TO Product, purchase_date DATETIME, quantity INT)
+CREATE EDGE Return_Event (FROM Person, TO Product, return_date DATETIME, reason STRING)
+----
+
+=== Indexing - Optimizing Searches
+
+Efficient querying relies on optimizing how the data is indexed. When you design a graph schema, proper indexing plays a key role in speeding up searches.
+
+==== Vertex Indexes
+
+Use vertex indexes for attributes that are queried frequently, like user ID or product ID.
+
+==== Secondary Indexes
+
+Use secondary indexes for attributes that require range queries or non-primary attributes.
```

Comment:
```suggestion
Use secondary indexes when the search behavior justifies having an index and either range queries are needed or a vertex index is not appropriate.
```
Documentation: https://docs.tigergraph.com/gsql-ref/4.2/ddl-and-loading/defining-a-graph-schema#_alter_index